rest-service/manager_rest/rest/resources_v3_1/deployments.py

```diff
@@ -192,6 +197,14 @@ def put(self):
             source_deployment=params[SOURCE_DEPLOYMENT],
             target_deployment=params[TARGET_DEPLOYMENT],
             created_at=now)
+        if hasattr(params[SOURCE_DEPLOYMENT], '_storage_id') and \
+                hasattr(params[TARGET_DEPLOYMENT], '_storage_id'):
+            source_id = str(params[SOURCE_DEPLOYMENT]._storage_id)
+            target_id = str(params[TARGET_DEPLOYMENT]._storage_id)
+            get_resource_manager().assert_no_cyclic_dependencies(source_id,
+                                                                 target_id)
+            get_resource_manager().modify_dependencies_graph(
+                source_id, target_id, 'add')
```

Comment:
Maybe the function `add dependency` should include the assertion for cyclic dependencies?

Reply:
the case in updating, when I first remove and then add, requires me to check for cycles before removing, so no gain here